go.mod

```diff
@@ -13,16 +15,16 @@ replace (
 
 require (
 	cloud.google.com/go v0.38.0
-	github.com/coreos/etcd v3.3.17+incompatible
 	github.com/dustinkirkland/golang-petname v0.0.0-20190613200456-11339a705ed2
 	github.com/go-logr/logr v0.1.0
 	github.com/google/go-cmp v0.3.0
 	github.com/google/gofuzz v1.0.0
 	github.com/robfig/cron/v3 v3.0.0
 	github.com/stretchr/testify v1.4.0
-	go.etcd.io/etcd v3.3.17+incompatible
+	// Pin spesific etcd version via tag. See https://github.com/etcd-io/etcd/pull/11477
+	go.etcd.io/etcd v0.0.0-20191023171146-3cf2f69b5738
```

Comment:
iirc these get re-written by `go mod tidy` anyway - if you just had the commit ref here it should work (and `go mod` will automatically compute to `v0.0.0-*` version of the ref)

Reply:
> These get re-written by `go mod tidy`

It doesn't seem to. I've run tidy a few times since I added this with no change.

I'm pretty much following the [documentation](https://github.com/etcd-io/etcd/blob/94f5b1ccb9b5a3e78e5ae4717037780452cd4a52/hack/get-usable-mod-version-from-tag/README.md) for the version workaround.

Reply:
What I mean is if you added/manually specified the *commit ref* here, `go mod tidy` would rewrite it into the form you've written here 😄 

I don't think using that linked script is necessary.. simply getting the ref for that tag and adding it here, then running `go mod tidy` should do it.